protocols/hyprland-surface-v1.xml

```diff
@@ -96,5 +96,17 @@
       <entry name="no_surface" value="0" summary="wl_surface was destroyed"/>
       <entry name="out_of_range" value="1" summary="given opacity was not in the range 0.0 - 1.0 (inclusive)"/>
     </enum>
+
+    <request name="set_visible_region" since="2">
+      <description summary="set the visible region of the surface">
+        This is an optimization hint that allows the compositor to avoid rendering
+        any part of the surface that does not fall into the visible region.
+        A NULL wl_region removes the hint.
```

Comment:
you couldn't have written it any more vaguely. Maybe just say it's akin to "opaque_region" of wl_surface except here it marks a region that the app does not render anything to (leaving it fully transparent)